### DISCUSSION TITLE
How can i get reproducibility?

### DISCUSSION THREAD

**Discussion by davidlambda256** [pyg-team/pytorch_geometric#6013] [Q&A]
I applied this code but cannot get reproducibility in pytorch_geometric..

`torch.manual_seed(random_seed)
torch.cuda.manual_seed(random_seed)
torch.cuda.manual_seed_all(random_seed) # if use multi-GPU
torch.backends.cudnn.deterministic = True
torch.backends.cudnn.benchmark = False
np.random.seed(random_seed)
random.seed(random_seed)`

How can i fix it?

**Comment by EdisonLeeeee** [comment]
If you were using `edge_index` as input, it is suggested to use `SparseTensor` instead for reproducibility. FYI: https://github.com/pyg-team/pytorch_geometric/issues/2788#issuecomment-870502307
